File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/adme/AdmeAnthropomorphicMovementInfoParser.java
package com.shmedo.configlibrary.iot.cmd.parser.adme;

import android.text.TextUtils;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser;
import com.shmedo.configlibrary.iot.model.adme.AdmeAnthropomorphicMovementInfo;

import java.util.HashMap;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2022/1/25 <br/>
 * 描述：     TODO
 */
public class AdmeAnthropomorphicMovementInfoParser implements IOTResultParser<AdmeAnthropomorphicMovementInfo> {
    @Override
    public AdmeAnthropomorphicMovementInfo parse(String result) {
        AdmeAnthropomorphicMovementInfo info = new AdmeAnthropomorphicMovementInfo();
        try {
            String[] keyValues = result.split("&");
            HashMap<String, String> keyValueMap = new HashMap<>();

            for (String keyValue : keyValues) {
                String[] strs = keyValue.split("=");
                if (strs.length < 2) {
                    keyValueMap.put(strs[0], "");
                } else {
                    keyValueMap.put(strs[0], strs[1]);
                }
            }
            info.setMode(TextUtils.isEmpty(keyValueMap.get("mode")) ? "" : keyValueMap.get("mode"));

            return info;
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    @Override
    public void validate(String result) {

    }

    @Override
    public IOTCommandType commandType() {
        return IOTCommandType.ADME_MD_GET_ANTHROPOMORPHIC_MOVEMENT_MODE;
    }
}

File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/parser/adme/AdmeAnthropomorphicMovementInfoParser.kt
package com.shmedo.configlibrary.iot.cmd.parser.adme

import com.shmedo.configlibrary.iot.enums.IOTCommandType
import com.shmedo.configlibrary.iot.interfaces.IOTResultParser
import com.shmedo.configlibrary.iot.model.adme.AdmeAnthropomorphicMovementInfo

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2022/1/25 <br></br>
 * 描述：     TODO
 */
class AdmeAnthropomorphicMovementInfoParser : IOTResultParser<AdmeAnthropomorphicMovementInfo?> {
    override fun parse(result: String): AdmeAnthropomorphicMovementInfo? {
        val info = AdmeAnthropomorphicMovementInfo()
        return try {
            val keyValues = result.split("&").toTypedArray()
            val keyValueMap = HashMap<String, String>()
            for (keyValue in keyValues) {
                val strs = keyValue.split("=").toTypedArray()
                if (strs.size < 2) {
                    keyValueMap[strs[0]] = ""
                } else {
                    keyValueMap[strs[0]] = strs[1]
                }
            }
            info.mode = keyValueMap.getOrDefault("mode", "NullKey")
            info
        } catch (ex: Exception) {
            ex.printStackTrace()
            null
        }
    }

    override fun validate(result: String) {}
    override fun commandType(): IOTCommandType {
        return IOTCommandType.ADME_MD_GET_ANTHROPOMORPHIC_MOVEMENT_MODE
    }
}